### ISSUE TITLE
Bug in Style/SymbolProc?

### ISSUE THREAD

**Issue by bquorning** [bbatsov/rubocop#2217]
In the RSpec documentation it is described [how a stub can yield to the caller's block](http://www.relishapp.com/rspec/rspec-mocks/v/3-3/docs/configuring-responses/block-implementation#yield-to-the-caller's-block):

Taking that code and removing the argument to `#call` results in this code:

```rb
# encoding: utf-8
RSpec.describe 'When the caller passes a block' do
  it 'can be yielded to from your implementation block' do
    dbl = double
    allow(dbl).to receive(:foo) { |&block| block.call }
    expect { |probe| dbl.foo(&probe) }.to yield_control
  end
end
```

Running RuboCop v0.34.0 gives me this offense:
```
test.rb:5:19: C: Pass &:call as an argument to receive instead of a block.
    allow(dbl).to receive(:foo) { |&block| block.call }
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

RuboCop is happy if I change `receive(:foo) { |&block| block.call }` into `receive(:foo, &:call)`, but it doesn’t work. I guess the cop doesn’t take the `&` in `|&block|` into consideration.

cc @lumeet
